packages/client/src/objectSet/createObjectSet.ts

```diff
@@ -63,20 +63,30 @@ export function createObjectSet<Q extends ObjectOrInterfaceDefinition>(
     // }) => {
     //   throw "TODO";
     // },
-    aggregateOrThrow: async <
+    aggregateOrThrow: (async <
       // AC extends AggregationClause<Q>,
       // GBC extends GroupByClause<O, K>,
       AO extends AggregateOpts<Q>,
     >(
-      req: AO,
+      req: AO & {
+        select:
+          & Pick<
+            AO["select"],
+            keyof AggregateOpts<Q>["select"] & keyof AO["select"]
+          >
+          & Record<
+            Exclude<keyof AO["select"], keyof AggregateOpts<Q>["select"]>,
+            never
+          >;
+      },
     ): Promise<AggregationsResults<Q, AO>> => {
       return aggregateOrThrow<Q, AO>(
         clientCtx,
         objectType,
         objectSet,
         req,
       );
-    },
+    }) as any,
```

Comment:
Why do we need the as any here?

Reply:
I can try to take it out but the types got so complex here that it was erroring that two identical types didnt match.

Reply:
Ah probably fine to just keep it then, things work so yay